src/ControlSystem/WriteData.hpp

```diff
@@ -157,31 +109,26 @@ void write_components_to_disk(
   // ControlSystems/Translation/Y.dat
   // ControlSystems/Translation/Z.dat
   const size_t num_components = function_at_current_time[0].size();
+  const auto& legend = WriterHelper::legend;
```

Comment:
Just inline the legend here and delete the `WriterHelper`

Reply:
Oh I meant: just write it right here in line 112, and `std::move` it into the simple action invocation below. It's taken by value by the action anyway.

Reply:
Ah ok my bad I misunderstood. Thanks for clarifying :)